### PR TITLE
Feature dropdown mode for flexberry-lookup component

### PR DESCRIPTION
**Pull request by yzaripova**

### CODE REVIEW

File: addon/components/flexberry-dropdown.js [hunk 1]
```diff
@@ -51,6 +51,15 @@ export default FlexberryBaseComponent.extend({
   tagName: '',
 
   /**
+   * Classes for component wrapper.
+   *
+   * @property classNames
```
Comment (AlexanderElyseev) [inline addon/components/flexberry-dropdown.js]: class or classNames?

File: addon/components/flexberry-dropdown.js [hunk 2]
```diff
@@ -181,6 +190,7 @@ export default FlexberryBaseComponent.extend({
 
     // Initialize properties which defaults could be defined in application configuration.
     this.initProperty({ propertyName: 'defaultText', defaultValue: null });
+
```
Comment (AlexanderElyseev) [inline addon/components/flexberry-dropdown.js]: empty line

File: addon/components/flexberry-lookup.js [hunk 3]
```diff
@@ -60,7 +60,6 @@ var FlexberryLookup = FlexberryBaseComponent.extend({
   value: undefined,
```
Comment (AlexanderElyseev) [inline addon/components/flexberry-lookup.js]: required yuidoc coments

File: addon/components/flexberry-lookup.js [hunk 4]
```diff
@@ -105,68 +104,68 @@ var FlexberryLookup = FlexberryBaseComponent.extend({
...
    * @default undefined
    */
-  autocompleteQueryOptions: undefined,
+  queryOptions: undefined,
 
   /**
    * Action's name to update model's relation value.
    *
-   * @property autocompleteUpdate
+   * @property updateLookupValue
```
Comment (AlexanderElyseev) [inline addon/components/flexberry-lookup.js]: different names

File: addon/components/flexberry-lookup.js [hunk 5]
```diff
@@ -211,110 +253,148 @@ var FlexberryLookup = FlexberryBaseComponent.extend({
...
+      this.set('autocompleteValue', this.get('value'));
+
+      this.$().search({
+        apiSettings: {
+          url: chooseRemoteData.url,
+          beforeSend: function(settings) {
+            let urlOptions = _this.get('queryOptions')(
+              {
+                relationName: chooseRemoteData.relationName,
+                lookupLimitFunction: chooseRemoteData.limitFunction,
```
Comment (AlexanderElyseev) [inline addon/components/flexberry-lookup.js]: We need to use new query language here.

File: addon/components/flexberry-lookup.js (hunk 4 above)
Comment (yzaripova) [inline addon/components/flexberry-lookup.js]: Yes. So what?

File: addon/components/flexberry-dropdown.js (hunk 1 above)
Comment (yzaripova) [inline addon/components/flexberry-dropdown.js]: Class. I corrected.

File: addon/components/flexberry-dropdown.js (hunk 2 above)
Comment (yzaripova) [inline addon/components/flexberry-dropdown.js]: I corrected.

File: addon/components/flexberry-lookup.js (hunk 4 above)
Comment (AlexanderElyseev) [inline addon/components/flexberry-lookup.js]: updateLookupAction or updateLookupValue?

**Pull request merged**